tldw_Server_API/app/core/DB_Management/media_db/legacy_content_queries.py

```
@@ -0,0 +1,93 @@
+    if not safe_media_ids:
+        return {}
+
+    keywords_map = {media_id: [] for media_id in safe_media_ids}
+    placeholders = ",".join("?" * len(safe_media_ids))
+    order_expr = db_instance._keyword_order_expression("k.keyword")  # type: ignore[attr-defined]
+    query = (
+        f"SELECT mk.media_id, k.keyword FROM MediaKeywords mk "  # nosec B608
+        "JOIN Keywords k ON mk.keyword_id = k.id "
+        "JOIN Media m ON mk.media_id = m.id "
+        f"WHERE mk.media_id IN ({placeholders}) AND k.deleted = ? AND m.deleted = ? "
+        f"ORDER BY mk.media_id, {order_expr}"
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)